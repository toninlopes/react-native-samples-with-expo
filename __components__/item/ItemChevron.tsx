import React from 'react';
import { View, StyleSheet } from 'react-native';
import { Ionicons } from '@expo/vector-icons';

const ItemChevron: React.FC = () => {
  return (
    <View style={styles.view}>
      <Ionicons name='chevron-forward-outline' size={28}/>
    </View>
  );
}

const styles = StyleSheet.create({
  view: {
    // paddingHorizontal: 12,
  },
});

export default ItemChevron;