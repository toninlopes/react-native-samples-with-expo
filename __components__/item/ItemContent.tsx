import React from "react";
import { Text, StyleSheet, View } from "react-native";
import type { ItemContentProps } from './types';

const ItemContent: React.FC<ItemContentProps> = ({ title }) => {
  return (
    <View style={styles.view}>
      <Text style={styles.title}>{title}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  view: {
    flex: 1,
  },
  title: {
      fontSize: 22,
  },
});

export default ItemContent;