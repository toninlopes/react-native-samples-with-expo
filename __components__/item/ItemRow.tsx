import React from "react";
import { StyleSheet, View } from "react-native";
import type { ItemRowProps } from './types';

const ItemRow: React.FC<ItemRowProps> = ({ children }) => {
  return (
    <View style={styles.row}>
      {children}
    </View>
  );
};

const styles = StyleSheet.create({
  row: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingVertical: 20,
  },
});

export default ItemRow;