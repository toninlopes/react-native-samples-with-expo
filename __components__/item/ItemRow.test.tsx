import React from 'react';
import renderer from 'react-test-renderer';
import ItemRow from './ItemRow';
import ItemChevron from './ItemChevron';
import ItemContent from './ItemContent';

describe('ItemRow', () => {
  test('When there is title as child.', () => {
    const itemRow = renderer.create((
      <ItemRow>
        <ItemContent title='Title' />
      </ItemRow>
    )).toJSON();
    expect(itemRow).toMatchSnapshot();
  });

  test('When there are title and chevron.', () => {
    const itemRow = renderer.create((
      <ItemRow>
        <ItemContent title='Title' />
        <ItemChevron />
      </ItemRow>
    )).toJSON();
    expect(itemRow).toMatchSnapshot();
  });
});