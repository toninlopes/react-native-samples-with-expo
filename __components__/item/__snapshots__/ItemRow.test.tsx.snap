// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemRow When there are title and chevron. 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "paddingVertical": 20,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 22,
        }
      }
    >
      Title
    </Text>
  </View>
  <View
    style={{}}
  >
    <Text />
  </View>
</View>
`;

exports[`ItemRow When there is title as child. 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "paddingVertical": 20,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 22,
        }
      }
    >
      Title
    </Text>
  </View>
</View>
`;
